### PR TITLE
Allow modules etc to provide Nightwatch tests

### PR DESCRIPTION
**Pull request by justafish**

### CODE REVIEW

File: core/tests/Drupal/Nightwatch/nightwatch.conf.js [hunk 1]
```diff
@@ -1,5 +1,27 @@
+import path from 'path';
+import glob from 'glob';
+
+const regex = /(.+\/tests\/src\/Nightwatch\/Tests)\/.*/g;
+const folders = {};
+let m;
+glob
+  .sync("**/tests/src/Nightwatch/Tests/**/*.js", {
```
Comment (dawehner) [inline core/tests/Drupal/Nightwatch/nightwatch.conf.js]: Nice!

File: core/tests/Drupal/Nightwatch/nightwatch.conf.js [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+    cwd:  path.resolve(process.cwd(), '..'),
+  })
+  .forEach(file => {
```
Comment (dawehner) [inline core/tests/Drupal/Nightwatch/nightwatch.conf.js]: This could be a reduce instead of a forEach

File: core/tests/README.md [hunk 3]
```diff
@@ -108,5 +108,8 @@ sudo -u www-data -E ./vendor/bin/phpunit -c core --testsuite functional-javascri
 - Install [Node.js](https://nodejs.org/en/download/) and [yarn](https://yarnpkg.com/en/docs/install). The versions required are specificed inside core/package.json in the `engines` field
 - Install [Google Chrome](https://www.google.com/chrome/browser/desktop/index.html)
 - Inside the `core` folder, run `yarn install`
-- Configure the nightwatch settings by copying `.example.env` to `.env` and editing as necessary.
+- Configure the nightwatch settings by copying `.env.example` to `.env` and editing as necessary.
 - Again inside the `core` folder, run `yarn test:nightwatch` to run the tests. By default this will output reports to `core/reports`
+- Nightwatch will run tests for core, as well as contrib and custom modules and themes. It will search for tests located under folders with the pattern `tests/src/Nightwatch/Tests`
+- To run only core tests, run `yarn test:nightwatch --tag core`
```
Comment (dawehner) [inline core/tests/README.md]: It would be nice to allow running a single test. This is super useful when fixing failures.

Comment (justafish) [reply core/tests/README.md]: @dawehner that's possible - would a link to http://nightwatchjs.org/guide#running-tests be sufficient here?

Comment (dawehner) [reply core/tests/README.md]: Works for me

File: core/tests/README.md [hunk 4]
```diff
@@ -108,5 +108,8 @@ sudo -u www-data -E ./vendor/bin/phpunit -c core --testsuite functional-javascri
...
+- To skip running core tests, run `yarn test:nightwatch --skiptags core`
```
[the remark this answers is not in the document]
Comment (ryanaslett) [reply core/tests/README.md]: oh, maybe all I need is the `--group` flag

[the remark this answers is not in the document]
Comment (justafish) [reply core/tests/README.md]: We could pass through `--tag modulename` as default, and allow that to be overridden if the contrib maintainer wants to run additional tests

[the remark this answers is not in the document]
Comment (dawehner) [reply core/tests/README.md]: For me as a module developer I would have expect to be able to say: run all the tests in a folder. Is there any way to provide this capability?

File: core/tests/README.md (hunk 3 above)
Comment (mglaman) [reply core/tests/README.md]: Is it just like PHPUnit `./bin/phpunit -c CONFIG web/modules/contrib/commerce_braintree/tests/src/Nightwatch/Tests/integration.js`?

File: core/tests/Drupal/Nightwatch/nightwatch.conf.js (hunk 2 above)
Comment (justafish) [reply core/tests/Drupal/Nightwatch/nightwatch.conf.js]: @dawehner can you provide an example? I'm not trying to reduce this to 1 value so not sure why that would work here.

File: core/tests/Drupal/Nightwatch/nightwatch.conf.js [hunk 5]
```diff
@@ -1,5 +1,28 @@
+import path from 'path';
+import glob from 'glob';
+
+const regex = /(.+\/tests\/src\/Nightwatch\/Tests)\/.*/g;
+let m;
+const folders = glob
+  .sync("**/tests/src/Nightwatch/Tests/**/*.js", {
```
Comment (dawehner) [inline core/tests/Drupal/Nightwatch/nightwatch.conf.js]: In a follow up we could maybe try to exclude folders like node_modules which could be super slow to scan.

**Pull request merged**